net/packet/packet_test.go

```diff
@@ -29,6 +30,52 @@ const (
 	Fragment = ipproto.Fragment
 )
 
+func fullHeaderChecksumV4(b []byte) uint16 {
+	s := uint32(0)
+	for i := 0; i < len(b); i += 2 {
+		if i == 10 {
+			// Skip checksum field.
+			continue
+		}
+		s += uint32(binary.BigEndian.Uint16(b[i : i+2]))
+	}
+	for s>>16 > 0 {
+		s = s&0xFFFF + s>>16
+	}
+	return ^uint16(s)
+}
+
+func TestHeaderChecksums(t *testing.T) {
+	// This is not a good enough test, because it doesn't
+	// check the various packet types or the many edge cases
+	// of the checksum algorithm. But it's a start.
+
+	tests := []struct {
+		name   string
+		packet []byte
+	}{
+		{
```

Comment:
Here's a TCP SYN (connect) to mon:

```
00004500003c542940004006b1ac6442d4336461980fb19401bb0a51ce7c00000000a002fbe038f60000020404d80402080a862bccd50000000001030307
```

Again, trim the first 2 bytes.

Reply:
Added